### ISSUE TITLE
generalizing travis.sh for linux/macos-based ci hosts

### ISSUE THREAD

**Issue by guillermooo** [randy3k/UnitTesting#71]
There are more SaaS CI servers today than a few years ago. While so far I've seen each CI environment can be adapted so that the `travis.sh` script  (and also `appveyor.ps1`) can be used with them, it should be possible to further decouple the `travis.sh` script from the Travis-CI specifics.

**Comment by randy3k** [comment]
Agree. That’s the reason I have separated the installing sublime text script. And we need to also make sure the refactoring won’t break the usages of existing packages.
